File: app/src/main/java/org/jeonfeel/moeuibit2/FavoriteDAO.java
package org.jeonfeel.moeuibit2;

import androidx.room.Dao;
import androidx.room.Query;

import java.util.List;

@Dao
public interface FavoriteDAO {

    @Query("SELECT * FROM Favorite")
    List<Favorite> getAll();

    @Query("INSERT INTO Favorite values(:market)")
    void insert(String market);

    @Query("DELETE FROM Favorite WHERE market = :market ")
    void delete(String market);

    @Query("SELECT * FROM Favorite WHERE market = :market")
    Favorite select(String market);

}

File: app/src/main/java/org/jeonfeel/moeuibit2/data/local/room/dao/FavoriteDAO.kt
package org.jeonfeel.moeuibit2.data.local.room.dao

import androidx.room.Dao
import androidx.room.Query
import org.jeonfeel.moeuibit2.data.local.room.entity.Favorite

@Dao
interface FavoriteDAO {

    @Query("SELECT * FROM Favorite WHERE exchange = :exchange")
    suspend fun getAllByExchange(exchange: String): List<Favorite>?

    @Query("INSERT INTO Favorite(market, exchange) VALUES(:market, :exchange)")
    suspend fun insert(market: String?, exchange: String?)

    @Query("DELETE FROM Favorite WHERE market = :market AND exchange = :exchange")
    suspend fun delete(market: String?, exchange: String?)

    @Query("SELECT * FROM Favorite WHERE market = :market AND exchange = :exchange")
    suspend fun select(market: String?, exchange: String?): Favorite?

    @Query("DELETE FROM Favorite")
    suspend fun deleteAll()
}